pkg/apis/core/helper/helpers.go

```diff
@@ -152,6 +152,14 @@ func IsExtendedResourceName(name core.ResourceName) bool {
 	return !IsDefaultNamespaceResource(name)
 }
 
+// IsExtendedResourceNameForQuota returns true if the extended resource name
+// has the quota related resource prefix.
+func IsExtendedResourceNameForQuota(name core.ResourceName) bool {
+	// As overcommit is not supported by extended resources for now,
+	// only quota objects in format of "requests.resourceName" is allowed.
+	return !IsDefaultNamespaceResource(name) && strings.HasPrefix(string(name), core.DefaultResourceRequestsPrefix)
```

Comment:
>can we restrict extended resources from having a requests. prefix so that convention is restricted to quota?

Yes, things get complicated if the ER itself has a `requests.` prefix. So you think it reasonable to add it as an additional limitation to ER validation func `IsExtendedResourceName`?

> in addition, should we limit the len for an extended resource name so we know it can fit a quota resource name len check? we need to ensure that requests. can be prefixed to any extended name with this PR.

Good catch

By the way, as I implied [above](https://github.com/kubernetes/kubernetes/pull/57302#issuecomment-358539342), Changes I made here could not take effect, any suggestions about changes to the part?

Reply:
Another notable point is that func `IsExtendedResourceName` does not act as outward-facing validator, that is:
If a user creates a pod that requests ER whose name illegal(say exceed the length above), api-server would not explicitly refuse the creation, but inside components(device-plugin .etc) would not take control of the resource. Thus, user may not be aware of the error, and get confused by the behavior of the cluster.
What I could think of is to update extended resource documents accordingly, as a reminder.

Reply:
Can we extend validateContainerResourceName() to also call IsExtendedResourceName() for resources containing "/"?
+1 for also updating doc accordingly.

Reply:
@jiayingz I'm not quite sure about the issue of scope, that is, whether we could return an error directly if a user requests resources other than ER and resources in default namespace.

Reply:
Are you asking whether we should check explicitly on "kubernetes.io/xxx" resource name? Good question. I think the behavior should stay unchanged for non-ER resources. Roughly, I am thinking something like this in validateContainerResourceName():
if len(strings.Split(value, "/")) == 1 {
		if !helper.IsStandardContainerResourceName(value) {
			return append(allErrs, field.Invalid(fldPath, value, "must be a standard resource for containers"))
		}
} else if !IsDefaultNameSpaceResource(value) {
		if !helper.IsExtendedResourceName(value) {
			return append(allErrs, field.Invalid(fldPath, value, "doesn't follow extended resource name standard"))
		}
}

And we can extend IsExtendedResourceName(name) to check name string is a bit under the standard 253 k8s max name length to leave enough room for "requests." and "limits".

Reply:
Gotcha, thanks for your suggestion; 
and for the quota validation part `ValidateResourceQuotaResourceName`, let's remove the "/" check first as you suggested, to see if any problems in existing tests triggered; please let me know if you got some reason that we have to hold the check.
>And we can extend IsExtendedResourceName(name) to check name string is a bit under the standard 253 k8s max name length to leave enough room for "requests." and "limits".

The part is already in the PR, looking into func `ValidateResourceQuotaResourceName`, a quota item's max length is thought to be 63 though.

Reply:
Could you double check this? Taking a look at IsQualifiedName(), I think if the resource name contains domain name, i.e., in the format of domain/name, name length can't exceed 63, but the total length including domain and / is under 253 DNS1123SubdomainMaxLength.

Reply:
Oh, you are right

Reply:
So one way I could think of is to construct a string of "requests.NAME" in func `IsExtendedResourceName()`, and call IsQualifiedName() for a validation, but I'm not sure if it wise to import validation package here.

Seems we never validate the length in resource name check funcs, thus, we are faced with the risk that a resource whose name is of proper format but length beyond the limit, could be successfully declared (e.g. discoverd by kubelet), but failed to be consumed (failed to pass container resources validation), same things may happen to resources like hugepage. I'm not sure it's by design or a bug to fix.

Would like to hear you guys opinions.